src/app/_components/list-source-data-research-context.tsx

```diff
@@ -1,85 +1,62 @@
-        response: response,
-        researchContextID: props.researchContextID,
-      };
-      const controller = clientContainer.get<BrowserListSourceDataController>(CONTROLLERS.LIST_SOURCE_DATA_CONTROLLER);
-      await controller.execute(controllerParameters);
-      return response;
-    },
+    queryFn: querySources(setListSourceDataViewModel, props.researchContextID),
   });
 
   const downloadMutation = useMutation({
     mutationKey: ["download-source-data"],
-    retry: 3,
-    retryDelay: 3000,
-    mutationFn: async (params: {relativePath: string, sourceDataName: string}) => {
-      const signalFactory = signalsContainer.get<(initialValue: TFileDownloadViewModel, update?: (value: TFileDownloadViewModel) => void) => Signal<TFileDownloadViewModel>>(SIGNAL_FACTORY.KERNEL_FILE_DOWNLOAD);
-      const response: Signal<TFileDownloadViewModel> = signalFactory(
-        {
-          status: "request",
-        } as TFileDownloadViewModel,
-        setDownloadSourceDataViewModel,
-      );
-      const controllerParameters: TBrowserFileDownloadControllerParameters = {
-        response: response,
-        relativePath: params.relativePath,
-        localPath: params.sourceDataName,
-      };
-      const controller = clientContainer.get<BrowserFileDownloadController>(CONTROLLERS.KERNEL_FILE_DOWNLOAD_CONTROLLER);
-      await controller.execute(controllerParameters);
-      return response;
-    },
+    retry: DEFAULT_RETRIES,
+    retryDelay: DEFAULT_RETRY_DELAY,
+    mutationFn: downloadSourceMutation(setDownloadSourceDataViewModel),
   });
 
   const handleDownloadSourceData = (name: string, relativePath: string) => {
-    console.log("Download source data", relativePath, name);
-    downloadMutation.mutate({relativePath: relativePath, sourceDataName: name});
+    downloadMutation.mutate({ relativePath: relativePath, sourceFilename: name });
   };
 
-  const handleUploadSourceData: () => void = () => {console.log("")};
+  const { toast } = useToast();
+
+  useEffect(() => {
+    if (downloadSourceDataViewModel.status === "error") {
+      toast({
+        title: "Error downloading the file",
+        description: downloadSourceDataViewModel.message,
+        variant: "error",
+      });
+    }
+  }, [downloadSourceDataViewModel]);
 
   const isSourceDataLoading = listSourceDataViewModel.status === "request";
   const sourceData = listSourceDataViewModel.status === "success" ? listSourceDataViewModel.sourceData : [];
+  const errorOverlayProperties =
+    listSourceDataViewModel.status === "error"
+      ? {
+          errorStatus: true,
+          overlayText: listSourceDataViewModel.message,
+        }
+      : undefined;
+
   return (
-      <SourceDataAGGrid isLoading={isSourceDataLoading} isUploading={false} enableUpload={false} rowData={sourceData} handleDownloadSourceData={handleDownloadSourceData} handleUploadSourceData={handleUploadSourceData} />
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
```

Comment:
This strongly suggest a need for changing the interface of the table component. This will be addressed in https://github.com/dream-aim-deliver/rage-ui-kit/issues/45